spec/features/tag_tasks_index_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+describe 'タスクのタグからタグに紐づくタスクの一覧ページを表示する' do
+  before do
+    user
+    oauth_sign_in(auth_hash: auth_hash)
+  end
+
+  let(:user) { create_user_from_oauth_credential(auth_hash) }
+  let(:auth_hash) { generate_auth_hash }
+
+  def task_summary_by_order(order)
+    task_summary = all('.task-summary')[order - 1]
+    {
+      title: task_summary.first('.task-title').text,
+      tags: task_summary.all('.tag').map(&:text),
+    }
+  end
+
+  it do
+    create_task(user.id, 'タスクのタイトル3', nil, %w(タグ1))
+    create_task(user.id, 'タスクのタイトル2', nil, %w(タグ2))
+    create_task(user.id, 'タスクのタイトル1', nil, %w(タグ1))
+    visit root_path
+    first('.task-summary').click_link('タグ1')
+    tag = first('.header .tag').text
+    expect(tag).to eq('タグ1')
+  end
+
+  it do
+    task_c = create_task(user.id, 'タスクのタイトルC', nil, %w(タグ3 タグ1))
+    task_b = create_task(user.id, 'タスクのタイトルB', nil, %w(タグ2))
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `task_b`. Did you mean `task_c`?